doc/nrf/libraries/others/emds.rst

```diff
@@ -0,0 +1,131 @@
+Its intended use is for storing of data undergoing frequent changes during runtime.
+
+Implementation
+==============
+The :kconfig:option:`CONFIG_EMDS` Kconfig option enables the emergency data storage.
+
+The application must initialize the pre-allocated flash area by using the :c:func:`emds_init` function.
+The :kconfig:option:`CONFIG_EMDS_SECTOR_COUNT` option defines how many sectors should be used to store data.
+
+The allocated storage space must be larger than the combined data stored by the application.
+Allocating a larger flash area will demand more resources, but also increase the life expectancy of the flash.
+The chosen size should reflect the amount of data stored, the available flash resources, and how the application calls the :c:func:`emds_store` function.
+In general, it should not be necessary to allocate a large flash area, since only a limited set of data should be stored to ensure swift completion of writing the flash on shutdown.
+
+The memory location that is going to be stored must be added on initialization.
+All memory areas must be provided through entries containing an ID, data pointer, and data length, using the :c:func:`emds_entry_add` function and the :c:macro:`EMDS_STATIC_ENTRY_DEFINE` macro.
+Entries to be stored when the emergency data storage is triggered need their own unique IDs that are not changed after a reboot.
+
+When all entries are added, the :c:func:`emds_load` function restores the entries into the memory areas from the flash.
+
+After restoring the previous data, the application must run the :c:func:`emds_prepare` function to prepare the flash area for receiving new entries.
+If the remaining empty flash area is smaller than the required data size, the flash area will be automatically erased to increase the available flash area.
+
+The storage is done in deterministic time, so it is possible to know how long it takes to store all registered entries.
+However, this is chip-dependent, so it is important to measure the time. The `Nordic Semiconductor Infocenter`_ contains chip information and datasheet, and timing values can be found under the "Electical specification" for the Non-volatile memory controller.
+The following Kconfig options can be configured:
+
+* :kconfig:option:`CONFIG_EMDS_FLASH_TIME_BASE_OVERHEAD_US`
+* :kconfig:option:`CONFIG_EMDS_FLASH_TIME_ENTRY_OVERHEAD_US`
+* :kconfig:option:`CONFIG_EMDS_FLASH_TIME_BASE_OVERHEAD_US`
+
+When changing the :kconfig:option:`CONFIG_EMDS_FLASH_TIME_BASE_OVERHEAD_US` option, it is important that the worst time is considered.
+
+Also consider the time for erase when doing garbage collection in NVS.
+If partial erase is not enabled or supported, the time of a complete sector erase has to be included.
+When partial erase is enabled and supported by the hardware, include the time it takes for the scheduler to trigger, which is depending on the time defined in :kconfig:option:`CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE_MS`.
```

Comment:
`include the time it takes for the scheduler to trigger`: Is there any standardized way to measure this, or is this time available somewhere in some KConfig settings? Otherwise, this is quite vague and will generate devzone questions.

But: If this is complicated to fix now, we could improve this later through a separate task.

Reply:
This is a time the user can set in KConfig.